components/ApiKeyUsage.js

```diff
@@ -0,0 +1,90 @@
+import { Component } from 'react'
+import { Button } from 'react-bootstrap'
+import { withAuth } from 'use-auth0-hooks'
+
+import ApiKeyUsageChart from './ApiKeyUsageChart'
+
+/**
+ * Shows API Key usage for all API Gateway usage plans found in logs prop.
+ */
+class ApiKeyUsage extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      aggregatedView: false
+    }
+  }
+
+  _getEmptyLogsMessage = () => <p className='lead text-center'>No usage logs found</p>
+
+  _toggleAggregatedView = () => {
+    this.setState({aggregatedView: !this.state.aggregatedView})
+    // TODO: add query param?
+  }
+
+  render () {
+    const { isAdmin, logs, logsError } = this.props
+    const { aggregatedView } = this.state
+    const hasLogs = logs && logs.length > 0
+    if (!hasLogs) {
+      return this._getEmptyLogsMessage()
+    }
+    let charts
+    if (aggregatedView) {
+      // Collect all per key request counts into single plan.
+      let plan
+      logs.forEach((p, i) => {
+        if (!plan) plan = p
+        else plan.result.items = {...plan.result.items, ...p.result.items}
```

Comment:
This is modifying items within the `logs` property and causing multiple charts to show up when switch back to the individual keys view.